### PR TITLE
Delete CloudFormation stack on kops delete cluster

### PR DESCRIPTION
**Pull request by kris-nova**
Adding support to delete a cloud formation stack on `kops delete cluster`

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1947)

### CODE REVIEW

File: tests/integration/minimal/cloudformation.json [hunk 1]
```diff
@@ -538,7 +538,27 @@
...
             {
               "Action": [
-                "route53:*"
+                "route53:ChangeResourceRecordSets",
+                "route53:ListResourceRecordSets",
+                "route53:GetHostedZone"
+              ],
+              "Effect": "Allow",
+              "Resource": [
+                "arn:aws:route53:::hostedzone/Z1AFAKE1ZON3YO"
```
Comment (kris-nova) [inline tests/integration/minimal/cloudformation.json]: I think this actually has to go, this is probably something with my local environment getting in the way

**Comment by ajohnstone** [comment]
Wouldn't it be better to have a callback rather than a specific implementation. So if someone wanted to hook into the kops lifecycle they could. I solved by wrapping kops in a python script for example.

**Comment by kris-nova** [comment]
@ajohnstone I don't understand what you are suggesting here? Are you suggesting we start wrapping kops in Python? 😕

**Comment by kris-nova** [comment]
Talked with @ajohnstone and there was some miscommunication on the new `--target cloudformation` feature!

**Comment by kris-nova** [comment]
Where this feature is relevant in the following workflow:

```
export KOPS_NAME=demo.nivenly.com
kops create cluster --zones us-west-2a --target cloudformation --name $KOPS_NAME
aws cloudformation create-stack --capabilities CAPABILITY_NAMED_IAM --stack-name kubernetes-demo-nivenly-com --template-body file://out/cloudformation/kubernetes.json
kops delete cluster --name $KOPS_NAME --yes
```

**Comment by justinsb** [comment]
This looks good.  I'm just thinking about what we actually expect the lifecycle to be here.  I guess if you delete the cluster with kops, you do want to delete the CF resource as well, as otherwise I would expect CF would recreate it (?).  And we do have `--unregister` which means "remove from kops but leave in CF".  So I think this is right.  Was this more or less the same thing as you were thinking @kris-nova ?

**Pull request merged**